src/ui/hooks/utils.ts

```diff
@@ -16,6 +17,16 @@ export function pvStateSelector(
   return results;
 }
 
+export function deviceSelector(device: string, state: CsState): DType {
+  return state.deviceCache[device];
+}
+
+export function deviceComparator(before: DType, after: DType): boolean {
```

Comment:
I'm not so sure about this one. If I am correct, `deviceComparator()` will be called each time the store updates, then anything using `useDevice()` will be re-rendered each time the store updates.

I'm not certain, so I propose we leave it as it is and then we can check our understanding.

Reply:
Ahh interesting, my understanding was that redux passes in the previous value for a specific device (found using the deviceSelector first), and then just compares the old with the new. I'll have a look into it.